test/catch2/sub_level_error/unit/test_sub_level_error_drop_conflict.cpp

```diff
@@ -116,43 +154,59 @@ TEST_CASE("Test WT_CONFLICT_BACKUP and WT_CONFLICT_DHANDLE", "[sub_level_error_d
 /*
  * This test case covers EBUSY errors resulting from drop while a lock is held by another thread.
  */
-TEST_CASE("Test CONFLICT_SCHEMA_LOCK and CONFLICT_TABLE_LOCK", "[sub_level_error_drop_conflict]")
+TEST_CASE("Test conflicts with checkpoint/schema/table locks", "[sub_level_error_drop_conflict]")
 {
     std::string config = "key_format=S,value_format=S";
     WT_SESSION *session_a = NULL;
     WT_SESSION *session_b = NULL;
     WT_ERROR_INFO *err_info_a = NULL;
     WT_ERROR_INFO *err_info_b = NULL;
 
-    /* This section directly uses the pthread library which is POSIX-only, so skip on Windows. */
-#ifndef _WIN32
-    SECTION("Test CONFLICT_SCHEMA_LOCK")
+    connection_wrapper conn_wrapper = connection_wrapper(".", "create");
+    prepare_session_and_error(&conn_wrapper, &session_a, &err_info_a);
+    prepare_session_and_error(&conn_wrapper, &session_b, &err_info_b);
+    REQUIRE(session_a->create(session_a, URI, config.c_str()) == 0);
+
+    SECTION("Test CONFLICT_CHECKPOINT_LOCK")
     {
-        connection_wrapper conn_wrapper = connection_wrapper(".", "create");
-        prepare_session_and_error(&conn_wrapper, &session_a, &err_info_a);
-        prepare_session_and_error(&conn_wrapper, &session_b, &err_info_b);
-        REQUIRE(session_a->create(session_a, URI, config.c_str()) == 0);
+        /* Attempt to drop the table while another session holds the checkpoint lock. */
+        WT_SPINLOCK *checkpoint_lock = &S2C((WT_SESSION_IMPL *)session_a)->checkpoint_lock;
+        std::thread lock_thread(
+          [&]() { thread_function_hold_spinlock(session_b, checkpoint_lock); });
+        __wt_sleep(0, 500000);
```

Comment:
How long is this sleep? Maybe it's worth to just wait a second instead? Could you also add a comment on why we need to wait?